### DISCUSSION TITLE
Como setar o CSS max-width de uma imagem igual ao natural width da imagem? Sem JS

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#1712] [Perguntas e Respostas]
Oi pessoal, tenho uma imagem centralizada na tela, e gostaria que o max-width fosse igual ao natural width da imagem, assim ela nunca vai ficar maior do que ela realmente é... alguém sabe se isso é possível sem JavaScript?

**Comment by renanlbl** [comment]
Irmao, se você quer a imagem com tamanho natural, você não mexe em width, nem width nem max-width... Se ela tem 1920px de largura, ela sempre terá isso, não se preocupe, ela não aumentará

**Comment by wilsonneto-swe** [comment]
@renanlbl provavelmente ele está pensando em responsividade e vai precisar diminuir a imagem quando a tela for menor, ai não vai conseguir usar ela ao natural...

@felquis da uma olhada no natualWidth/height.

Já é suportado nos browsers atuais, mas já faz pensando em um fallback.

Especificação: https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-naturalwidth
Can I use: https://caniuse.com/#feat=img-naturalwidth-naturalheight

-----  
:relaxed: 
by: Wilson Neto  
[Linkedin](https://linkedin.com/in/wilsonnetobr/) - [Blog](http://wilsonneto.com.br) - [Github](https://github.com/wilsonneto-dev)

**Comment by valkervieira** [comment]
https://codepen.io/valkervieira/pen/mdJmQvy

Usando `max-width: 100%;` caso o container seja menor do que a imagem ela vai ser redimensionada, caso seja maior ela vai ficar do tamanho natural máximo. No pen acima redimensiona a janela de exibição pra ver se é o que você quer.

**Comment by felquis** [comment]
https://github.com/frontendbr/forum/issues/1712#issuecomment-592586536 o problema aqui é que naturalWidth está apenas no JavaScript por enquanto.

https://github.com/frontendbr/forum/issues/1712#issuecomment-592596932 já aqui o demo funciona como esperado, porém `max-width: 100%` seria bem mais "elegante" se fosse `max-width: natural-width`. Será que tem alguma discussão sobre isso?

**Comment by wilsonneto-swe** [comment]
@borghettinho siim, verdade. @felquis mas o max-width = natural with não seria responsivo sózinho, e se usado com width: 100% seria o mesmo que max-width: 100%, particularmente prefiro assim... 

-----  
:relaxed: 
by: Wilson Neto  
[Linkedin](https://linkedin.com/in/wilsonnetobr/) - [Blog](http://wilsonneto.com.br) - [Github](https://github.com/wilsonneto-dev)
